### ISSUE TITLE
Provide the C# 8 nullable attributes as a NuGet package

### ISSUE THREAD

**Issue by tndata** [dotnet/corefx#40947]
In C# 8 (.NET Core 3) you added new nullability-attributes to allow us to manually annotate types with their nullability.

Located here:
roslyn/src/Workspaces/Core/Portable/Utilities/NullableHelpers/NullableAttributes.cs

Why not provide these attributes as a stand-alone NuGet package, so that legacy packages (.NET Core <2.x / .NET Standard <2.1) libraries can be annotated with these attributes as well.

Benefits:
* We can start annotating our legacy libraries 
* We would get more nullability warnings in our .NET Core 3 applications

The alternative is to manually add these attributes to my legacy libraries, but that feels a bit dirty. The nice thing is that the C# 8 compiler uses these attributes when used in legacy libraries.

**Comment by stephentoub** [comment]
cc: @terrajobst, @cartermp

**Comment by terrajobst** [comment]
It has come up. I'll see what we can do post .NET Core 3.0.

**Comment by sharwell** [comment]
@terrajobst It would help in the interim if the file could be provided under a public domain license, or somehow intentionally excluded from the current attribution requirements.

**Comment by terrajobst** [comment]
> @terrajobst It would help in the interim if the file could be provided under a public domain license, or somehow intentionally excluded from the current attribution requirements. It's easy to create a package that embeds the attribute definitions with `internal` accessibility for use, but the licensing situation is not clear when the embedded definitions are distributed.

I disagree. The MIT license is permissive any most real world software will already have to provide a TPN files b/c OSS is mainstream enough. One off licenses are expensive for us to maintain and will generally cause more confusion than whatever minor gains are to be had IMHO.

**Comment by chucker** [comment]
This package seems to roughly fulfill the requirements: https://www.nuget.org/packages/Nullable/

**Comment by jnm2** [comment]
@sharwell's https://github.com/tunnelvisionlabs/ReferenceAssemblyAnnotator provides source-defined internal attributes (useful for public types) AND rewrites the netfx/netstandard/old netcore reference assembles by copying attributes into them from matching APIs in .NET Core 3.
